### ISSUE TITLE
Throttle workers when everything is failing

### ISSUE THREAD

**Issue by annawoodard** [matz-e/lobster#436]
Idea from @dthain:

@WenzhaoLi had a run where close to 100% of jobs failing for some time due to a stale proxy file. In such situations it would be ideal for Lobster not to terminate itself, but rather throttle back tasks running until the failure rate improves (in case it is a transient issue).

**Comment by matz-e** [comment]
Good idea! Was this a user proxy? In that case we should still quit if the proxy expires…
